docker/src/main/java/brooklyn/entity/container/docker/DockerInfrastructureImpl.java

```diff
@@ -182,7 +182,7 @@ public String apply(@Nullable Entity input) {
                 .displayName("Docker Applications"));
 
         if (config().get(SDN_ENABLE) && config().get(SDN_PROVIDER_SPEC) != null) {
-            Entity sdn = addChild(EntitySpec.create(config().get(SDN_PROVIDER_SPEC))
+            Entity sdn = addChild(EntitySpec.create((EntitySpec<?>)config().get(SDN_PROVIDER_SPEC))
```

Comment:
Problem here is that without the cast what gets passed to the child is `org.apache.brooklyn.api.internal.AbstractBrooklynObjectSpec`. It's the `configure` method that's causing the problem but I don't currently see how/where to resolve it without the cast, previously `configure` was implemented in the `EntitySpec` and we didn't encounter this problem, but it now resides in `AbstractBrooklynObjectSpec` and we're losing the type along the way. I'm going to try to replicate what I'm seeing to try to understand it.